newa/cli.py

```diff
@@ -41,35 +41,55 @@
     )
 
 JIRA_NONE_ID = '_NO_ISSUE'
+STATEDIR_PARENT_DIR = '/var/tmp/newa'
+STATEDIR_NAME_PATTERN = '^run-([0-9]+)$'
 
 logging.basicConfig(
     format='%(asctime)s %(message)s',
     datefmt='%m/%d/%Y %I:%M:%S %p',
     level=logging.INFO)
 
 
-def default_state_dir() -> Path:
-    """ Returns the first unused directory matching /var/tmp/newa/run-[0-9]+ """
-    parent_dir = Path('/var/tmp/newa')
-    pattern = '^run-([0-9]+)$'
+def get_state_dir(use_ppid: bool = False) -> Path:
+    """ When not using ppid returns the first unused directory
+        matching /var/tmp/newa/run-[0-9]+
+        When using ppid searches for the latest state-dir directory
+        containing file $PPID.ppid
+    """
+    parent_dir = Path(STATEDIR_PARENT_DIR)
     counter = 0
+    ppid_filename = f'{os.getppid()}.ppid'
     try:
         obj = os.scandir(parent_dir)
-    except FileNotFoundError:
+    except FileNotFoundError as e:
+        if use_ppid:
+            raise Exception(f'{parent_dir} does not exist') from e
         return parent_dir / f'run-{counter}'
     for entry in obj:
-        r = re.match(pattern, entry.name)
+        r = re.match(STATEDIR_NAME_PATTERN, entry.name)
         if entry.is_dir() and r:
             c = int(r.group(1))
-            if c >= counter:
+            if use_ppid:
+                ppid_file = parent_dir / entry.name / ppid_filename
+                if ppid_file.exists() and c >= counter:
+                    counter = c
+            elif c >= counter:
                 counter = c + 1
-    return parent_dir / f'run-{counter}'
+    if counter > 0:
```

Comment:
`if counter:` is a more pythonic version, `0` is a false-ish value.

Reply:
fixed